src/Message/UniformCommunicateError.h

```diff
@@ -0,0 +1,47 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2020 QMCPACK developers.
+//
+// File developed by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Lab
+//
+// File created by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Lab
+//////////////////////////////////////////////////////////////////////////////////////
+#ifndef QMCPLUSPLUS_UNIFORMCOMMUNICATEERROR_H
+#define QMCPLUSPLUS_UNIFORMCOMMUNICATEERROR_H
+
+#include <exception>
+namespace qmcplusplus
+{
+/** This a subclass for runtime errors that will occur on all ranks.
+ *
+ *  This is intended to be thrown and caught within a single Communicate context.
+ *  
+ *  The intended use is with a catch block like this:
+ *
+ *  catch (const UniformMPIerror& ue)
+ *  {
+ *    my_local_comm->barrier_and_abort(ue.what());
+ *  }
+ *
+ *  Which insures that the error message actually makes it out on the head
+ *  rank before the MPI processes are aborted.
+ *  
+ *  If even one member of the comm i.e. my_local_comm does not experience the issue
+ *  you will HANG the job until the walltime expires.
+ *  
+ *  If you split a communicator it is your responsibility to add a try catch block in your
```

Comment:
I feel this is an unnecessary burden. `UniformCommunicateError` can be restricted to places where a communicator is not accessible and must be caught properly wherever a communicator is available. If we restrict the use case narrow to this. We don't need to worry about accidental hanging by mishandling of exceptions.

Reply:
It doesn't effect AFQMC since they don't use the error.  I'm happier with the status quo with std::run_time exception than assuming my_comm_ is in scope or having many try catch blocks that are just there because of fear.
If you want to put a split over existing code, or you want to use UniformCommunicateError under your split in new code you need to understand this.